src/Evolution/Systems/Ccz4/FiniteDifference/EnforceConstrainedEvolution.hpp

```diff
@@ -0,0 +1,64 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/Protocols/Mutator.hpp"
+#include "DataStructures/Tensor/EagerMath/Determinant.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/Ccz4/FiniteDifference/System.hpp"
+#include "Evolution/Systems/Ccz4/FiniteDifference/Tags.hpp"
+#include "Evolution/Systems/Ccz4/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace Ccz4::fd {
+/*!
+ * \brief If constrained_evolution is true, enforce
+ * the unit determinant constraint on the conformal spatial metric
+ * and traceless constraint on the ATilde
```

Comment:
Is `constrained_evolution` an option tag or how is it set? Can you expand the dox a bit to point to the appropriate tags and note if they are hard-coded or option tags?